### ISSUE TITLE
Duplicate blocks when using eager loading

### ISSUE THREAD

**Issue by jlachancekffein** [spicywebau/craft-neo#247]
For some unknown reason, the query result using the Neo plugin using the eager loading returns duplicate blocks.

We have always used the eager load in connection with Neo in all our projects without problem.

Here is an example of a problem using globals with a Neo field (search_bloc)

```
$global = \craft\elements\GlobalSet::find();

/*
search_bloc handle is the Neo field
search_bloc.bloc:commons_cta handle is a matrix field
search_bloc.bloc:commons_cta.internal:ctaSection is a entry field
*/

 $global
 ->handle('search')
 ->with([
            'search_bloc',
            'search_bloc.bloc:commons_cta',
            'search_bloc.bloc:commons_cta.internal:ctaSection'
]);

$global = $global->one();

/*  At this point, search_bloc is an array vs neoQuery because previous with() */
$blocks = $global->search_bloc;

/* Debug */
$debug = array_map(function ($bloc) {
            return [
            'id' => $bloc->id,
            'uid' => $bloc->uid,
            'siteId' => $bloc->siteId,
            'draftId' => $bloc->draftId,
            'revisionId' => $bloc->revisionId,
            'enabled' => $bloc->enabled,
            'archived' => $bloc->archived,
            'dateDeleted' => $bloc->dateDeleted
            ];
            return $bloc->id;
  }, $blocks);

 \Craft::dd($debug);

```

The output of the $debug variable :

```
[
    0 => [
        'id' => '12018'
        'uid' => '7d0051d6-09be-4e1c-a18e-abf0f053d06b'
        'siteId' => '1'
        'draftId' => null
        'revisionId' => null
        'enabled' => '1'
        'archived' => '0'
        'dateDeleted' => null
    ]
    1 => [
        'id' => '12018'
        'uid' => '7d0051d6-09be-4e1c-a18e-abf0f053d06b'
        'siteId' => '1'
        'draftId' => null
        'revisionId' => null
        'enabled' => '1'
        'archived' => '0'
        'dateDeleted' => null
    ]
    2 => [
        'id' => '12018'
        'uid' => '7d0051d6-09be-4e1c-a18e-abf0f053d06b'
        'siteId' => '1'
        'draftId' => null
        'revisionId' => null
        'enabled' => '1'
        'archived' => '0'
        'dateDeleted' => null
    ]
]
```

We can clearly see that the block is duplicate 3 times with the same ID.

The problem seems to have occurred recently.

Has the eagerloading system changed since a recent release?

Another problem of this kind has been detected: https://github.com/spicywebau/craft-neo/issues/243#issuecomment-522592986

#### Other information
- Neo version: 2.4.3
- Craft version: 3.2.10
- Is the affected Craft installation multi-site: yes
- (if multi-site) Are the affected Neo fields set to manage blocks on a per-site basis: no
- (if multi-site, > 2.4.0) What is the propagation method of the neo field: all

**Comment by pvldigital** [comment]
@jlachancekffein hmm I'm sure eager-loading still functions the same. I'll look into it after #243.

**Comment by pvldigital** [comment]
@jlachancekffein i've made a change on how eagerloading retrieves the blocks in my fork of neo: https://github.com/pvldigital/craft-neo

You can test it by cloning the repo changing the require for neo to `"spicyweb/craft-neo": "^2.4.4",` and add the path to the clone repo.
e.g:
```
{
   "type": "path",
   "url": "../../Plugins/craft-neo",
   "options": {
     "symlink": true
   }
},
```

**Comment by pvldigital** [comment]
@jlachancekffein v2.4.4 is now released. It should fix the issue with duplicated content like #243
